src/rendering/MaterialParser/MaterialParser.cc

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2023 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <iostream>
+#include <sstream>
+
+#include "ConfigLoader.hh"
+#include "MaterialParser.hh"
+
+using namespace gz;
+using namespace sim;
+
+/////////////////////////////////////////////////
+MaterialParser::MaterialParser()
+{
+  this->configLoader = new ConfigLoader(".material");
```

Comment:
`configLoader` can just be an object member of `MaterialParser` instead of a pointer and initialized during construction. I don't see a need to allocate it on the heap and using a destructor.